pkg/kubectl/cmd/login.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+)
+
+func NewCmdLogin(f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "login Log in to the cluster.",
+		Long:  `login obtains the neccesary credentials to make authenticated requests to the cluster. The type of log in depends on the 'auth-provider' type in your configuration, and will likely require some type of user interaction.`,
```

Comment:
> Isn't that case is supported by --user=XXX ? An AuthProvider belongs to a AuthInfo (i.e. within user in the .kube/config), they don't exist on their own in in the Config.

I don't think so.  That pre-supposes a .kube/config file that has a stubbed user stanza.  I don't think that the should be the assumption `kubectl login`.  I think `kubectl login` is where I go to get from nothing to working .kube/config in a single command.